src/temperature.rs

```diff
@@ -204,6 +208,35 @@ impl ::std::cmp::PartialOrd for Temperature {
     }
 }
 
+#[cfg(feature = "from_str")]
+impl FromStr for Temperature {
+    type Err = std::num::ParseFloatError;
+
+    /// Create a new Temperature from a string
+    /// Plain numbers in string are considered to be Celsius
+    fn from_str(val: &str) -> Result<Self, Self::Err> {
+        if val.is_empty() {
+            return Ok(Temperature::from_celsius(0.0));
+        }
+
+        let re = Regex::new(r"([0-9.]*)\s?(deg|\u{00B0}|)?\s?([fckrFCKR]{1})$").unwrap();
```

Comment:
```suggestion
        let re = Regex::new(r"\s*([0-9.]*)\s?(deg|\u{00B0}|)?\s?([fckrFCKR]{1})\s*$").unwrap();
```
How about supporting strings with leading/trailing spaces? It's probably a frequent error source.

Reply:
Yeah, good idea. Changed and committed!